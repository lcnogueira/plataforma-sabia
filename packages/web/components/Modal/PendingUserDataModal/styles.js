import styled, { css } from 'styled-components';

export const Modal = styled.div`
	display: flex;

	> div:first-child > img {
		width: 100%;
		height: 100%;
	}
`;

export const InfosContainer = styled.div`
	${({ theme: { colors } }) => css`
		display: flex;
		flex-direction: column;

		margin-left: 6.8rem;

		> p {
			color: ${colors.red};
			font-size: 2.8rem;
			font-weight: 500;
			margin-bottom: 0.8rem;
			line-height: 3.3rem;
		}

		> span {
			font-size: 1.6rem;
			line-height: 2.4rem;
		}
	`}
`;

export const Button = styled.button`
	${({ theme: { colors } }) => css`
		background: ${colors.primary};
		border: none;
		outline: none;

		align-self: flex-start;
		margin-top: auto;
		padding: 0.4rem 0.8rem;

		color: ${colors.white};
		text-transform: uppercase;
		font-weight: bold;
		font-size: 1.4rem;
		line-height: 2.4rem;

		:hover,
		:focus {
			background: ${colors.darkOrange};
		}
	`}
`;
