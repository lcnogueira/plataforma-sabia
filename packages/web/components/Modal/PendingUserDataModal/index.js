import React from 'react';
import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import { Modal, Button, InfosContainer } from './styles';

const PendingUserDataModal = ({ closeModal, message }) => {
	const router = useRouter();

	const handleClick = () => {
		closeModal();
		router.push('/user/my-account');
	};

	return (
		<Modal>
			<div>
				<img
					src="/alone-rafiki.svg"
					alt="Rapaz sentado em um balanço e ao lado um balanço vazio representando solidão"
				/>
			</div>

			<InfosContainer>
				<p>Você possui dados pendentes</p>
				<span>{message}</span>

				<Button onClick={handleClick}>Ir para meu perfil</Button>
			</InfosContainer>
		</Modal>
	);
};

PendingUserDataModal.propTypes = {
	closeModal: PropTypes.func.isRequired,
	message: PropTypes.string.isRequired,
};

export default PendingUserDataModal;
