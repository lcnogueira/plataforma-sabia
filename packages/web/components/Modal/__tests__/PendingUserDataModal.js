import React from 'react';
import * as Router from 'next/router';

import { render, screen, fireEvent } from 'test-utils';
import PendingUserDataModal from '../PendingUserDataModal';

const closeModalMock = jest.fn();

describe('<PendingUserDataModal />', () => {
	it('should render correctly', () => {
		const { container } = render(
			<PendingUserDataModal
				closeModal={closeModalMock}
				message="Complete o seu cadastro para..."
			/>,
		);

		expect(screen.getByRole('img')).toBeInTheDocument();
		expect(screen.getByRole('button')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should closeModal and redirect to /user/my-account when clicking go to profile button', () => {
		jest.spyOn(Router, 'useRouter').mockReturnValue({
			push: jest.fn(),
		});

		render(
			<PendingUserDataModal
				closeModal={closeModalMock}
				message="Complete o seu cadastro para..."
			/>,
		);

		fireEvent.click(screen.getByRole('button'));

		expect(Router.useRouter().push).toHaveBeenCalledWith('/user/my-account');
		expect(Router.useRouter().push).toHaveBeenCalledTimes(1);
		expect(closeModalMock).toHaveBeenCalledTimes(1);
	});
});
