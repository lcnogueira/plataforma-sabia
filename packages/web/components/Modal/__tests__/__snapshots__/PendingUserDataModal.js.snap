// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PendingUserDataModal /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > div:first-child > img {
  width: 100%;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 6.8rem;
}

.c1 > p {
  color: hsla(4,85%,57%);
  font-size: 2.8rem;
  font-weight: 500;
  margin-bottom: 0.8rem;
  line-height: 3.3rem;
}

.c1 > span {
  font-size: 1.6rem;
  line-height: 2.4rem;
}

.c2 {
  background: hsla(29,94%,62%);
  border: none;
  outline: none;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-top: auto;
  padding: 0.4rem 0.8rem;
  color: hsla(0,0%,100%);
  text-transform: uppercase;
  font-weight: bold;
  font-size: 1.4rem;
  line-height: 2.4rem;
}

.c2:hover,
.c2:focus {
  background: hsla(29,73%,56%);
}

<div>
  <div
    class="c0"
  >
    <div>
      <img
        alt="Rapaz sentado em um balanço e ao lado um balanço vazio representando solidão"
        src="/alone-rafiki.svg"
      />
    </div>
    <div
      class="c1"
    >
      <p>
        Você possui dados pendentes
      </p>
      <span />
      <button
        class="c2"
      >
        Ir para meu perfil
      </button>
    </div>
  </div>
</div>
`;
